### PR TITLE
Sort languages when writing XMP metadata for reproducible builds

### LINKED ISSUE #3467: Builds are not always reproducible due to document language detection
### Description

MRE

```typ
#set document(date: none)
#show "1234": set text(lang: "de")

asdf 1234
```

```sh
typst c main.typ
xxd main.pdf > 1.hex
typst c main.typ
# wait different sha256sum
xxd main.pdf > 2.hex
diff 1.hex 2.hex
```

```diff
668,669c668,669
< 000029b0: 663a 6c69 3e64 653c 2f72 6466 3a6c 693e  f:li>de</rdf:li>
< 000029c0: 3c72 6466 3a6c 693e 656e 3c2f 7264 663a  <rdf:li>en</rdf:
---
> 000029b0: 663a 6c69 3e65 6e3c 2f72 6466 3a6c 693e  f:li>en</rdf:li>
> 000029c0: 3c72 6466 3a6c 693e 6465 3c2f 7264 663a  <rdf:li>de</rdf:
```

### Reproduction URL

_No response_

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

### PR DESCRIPTION
**Pull request by istudyatuni**
fix #3467

### CODE REVIEW

File: crates/typst-pdf/src/lib.rs [hunk 1]
```diff
@@ -233,7 +233,12 @@ fn write_catalog(ctx: &mut PdfContext, ident: Option<&str>, timestamp: Option<Da
     info.finish();
     xmp.num_pages(ctx.document.pages.len() as u32);
     xmp.format("application/pdf");
-    xmp.language(ctx.languages.keys().map(|lang| LangId(lang.as_str())));
+    xmp.language({
+        let mut languages: Vec<_> = ctx.languages.keys().collect();
```
Comment (istudyatuni) [inline crates/typst-pdf/src/lib.rs:237]: Not sure that collect and Vec is good

**Comment by laurmaedje** [comment]
Ah, there are actually two problems! This and the max_by_key. We should do the collect first and then the max by key on the vector. We could also use BTreeMap, but I would assume that it is more expensive than HashMap + sort.

**Comment by istudyatuni** [comment]
max_by_key is used to detect most used language?

**Comment by laurmaedje** [comment]
Yes

**Comment by istudyatuni** [comment]
> We should do the collect first and then the max by key on the vector

does we really need to sort first, if `max` will still return maximum? or am I missing something

**Comment by istudyatuni** [comment]
> That's why I thought it only happens if the number of text items is the same. Because then there is no unique maximum and it depends on the order.

Second parameter in tuple in max by key is lang, so languages sorted alphabetically

File: crates/typst-pdf/src/lib.rs [hunk 2]
```diff
@@ -233,7 +234,7 @@ fn write_catalog(ctx: &mut PdfContext, ident: Option<&str>, timestamp: Option<Da
     info.finish();
     xmp.num_pages(ctx.document.pages.len() as u32);
     xmp.format("application/pdf");
-    xmp.language(ctx.languages.keys().map(|lang| LangId(lang.as_str())));
+    xmp.language(ctx.languages.keys().map(|lang| LangId(lang.as_str())).rev());
```
Comment (istudyatuni) [inline crates/typst-pdf/src/lib.rs:237]: This is wrong

**Comment by istudyatuni** [comment]
example for

```rs
let lang = ctx
    .languages
    .iter()
    .rev()
    .max_by_key(|(_, &count)| count)
    .map(|(&l, _)| l);
```

```
iter.rev.max_by_key -> lang
(c, 3), (b, 2), (a, 3) -> a
(c, 1), (b, 2), (a, 3) -> a
(c, 2), (b, 2), (a, 1) -> b
```

**Comment by laurmaedje** [comment]
Why `rev` at all if we use `BTreeMap`? It only needs to be consistent after all, we don't really care whether it is the first or last language.

**Comment by istudyatuni** [comment]
> I would assume that it is more expensive than HashMap + sort

I think it won't be noticeable because it will be rare that a document will have >3 languages

**Comment by laurmaedje** [comment]
> Second parameter in tuple in max by key is lang, so languages sorted alphabetically

Yeah, I totally missed that. Sorry for all the confusion!

**Pull request merged**